torch/_refs/nn/functional/__init__.py

```diff
@@ -59,6 +60,43 @@
 
 Tensor = torch.Tensor
 
+
+@register_decomposition(torch.ops.aten.alpha_dropout)
+def alpha_dropout(
+    self: TensorLikeType, p: float = 0.5, training: bool = True, inplace: bool = False
+) -> TensorLikeType:
+
+    if inplace:
+        raise NotImplementedError
+
+    if not training:
+        return self
+
+    assert p <= 1
+    assert p >= 0
+
+    if p == 1:
+        return refs.zeros_like(self)
+
+    if p == 0:
+        return self
+
+    noise = refs.lt(
+        refs.uniform(
+            self.shape, low=0.0, high=1.0, dtype=torch.float32, device=self.device
+        ),
+        1 - p,
+    )
+
+    alpha = 1.7580993408473766
+
+    a = 1.0 / ((alpha * alpha * p + 1) ** 2 * (1 - p) ** 2)
```

Comment:
https://github.com/pytorch/pytorch/blob/6faa6c68e8b76fb68f3a2b2783685102d0e87c00/aten/src/ATen/native/Dropout.cpp#L79

Shouldn't this be calling `math.sqrt`?

Reply:
Updated